gwcs/tests/test_geometry.py

```diff
@@ -92,3 +114,76 @@ def test_cartesian_to_spherical_mixed_Q(cart_to_spher, x, y, z):
         cart_to_spher(x, y, z)
     assert (arg_err.value.args[0] == "All arguments must be of the same type "
             "(i.e., quantity or not).")
+
+
+@pytest.mark.parametrize('wrap_at', ['1', 180., True, 180j, [180], -180, 0])
+def test_c2s2c_wrong_wrap_type(spher_to_cart, cart_to_spher, wrap_at):
+    err_msg = "'wrap_lon_at' must be an integer number: 180 or 360"
+    with pytest.raises(ValueError) as arg_err:
+        geometry.SphericalToCartesian(wrap_lon_at=wrap_at)
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        spher_to_cart.wrap_lon_at = wrap_at
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        geometry.CartesianToSpherical(wrap_lon_at=wrap_at)
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        cart_to_spher.wrap_lon_at = wrap_at
+    assert arg_err.value.args[0] == err_msg
+
+
+def test_cartesian_spherical_asdf(tmpdir):
+    s2c0 = geometry.SphericalToCartesian(wrap_lon_at=360)
+    c2s0 = geometry.CartesianToSpherical(wrap_lon_at=180)
+
+    # asdf round-trip test:
+    assert_roundtrip_tree({'c2s': c2s0, 's2c': s2c0}, tmpdir)
```

Comment:
Sorry I didn't notice this before (and no need to change it unless you want to), but `assert_roundtrip_tree` has a parameter for defining additional custom checks on the tree:

```python
def additional_checks(round_tripped_file):
    c2s = round_tripped_file['c2s']
    s2c = round_tripped_file['s2c']
   
    # ...

assert_roundtrip_tree({'c2s': c2s0, 's2c': s2c0}, tmpdir, asdf_check_func=additional_checks)
```